packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  HttpRequest,
+  HttpResponse,
+  ApiLoggerInterface,
+  LoggerInterface,
+  Level,
+} from '../coreInterfaces';
+import { LoggingOptions } from './loggerOptions';
+import { NullLogger } from './nullLogger';
+
+export class ApiLogger implements ApiLoggerInterface {
+  private readonly _logger: LoggerInterface;
+
+  constructor(loggingOp: LoggingOptions) {
+    this._logger = loggingOp.logger ?? new NullLogger();
+  }
+
+  public logRequest(coreRequest: HttpRequest): void {
+    this._logger.log(
+      Level.Info,
+      'Request ',
+      { HttpMethod: coreRequest.method },
+      { Url: coreRequest.url }
```

Comment:
I don't undertand the reasoning for using multiple arguments, each of which is a object containing a single property. Is there some constraint here? Or could this be redesigned to use a single object containing all structured data that we need to log?

Reply:
redesigned it to a record that contains all infromation.